### ISSUE TITLE
Cannot create a single channel image

### ISSUE THREAD

**Issue by jice-nospam** [johannesvollmer/exrs#171]
# Description
When using a single channel image, the pixel function should return a (f32) which is as far as I understand simplified into f32 by the compiler. This results in a compilation error :
```
the trait bound `f32: IntoRecursive` is not satisfied
```

# Reproducing
```
let channels = SpecificChannels::build()
        .with_channel("H")
        .with_pixel_fn(|_| {
            (0.0f32)
        });
```

# Environment
- Windows 10
- EXRS Version 1.5.2

**Comment by johannesvollmer** [comment]
Thanks for taking the time :) 
yes, this is probably a design issue in ears, I will have a look soon

**Comment by johannesvollmer** [comment]
If you need a workaround, there are multiple options.

The easiest would be to use dynamic channels, which loads all the channels without the need to specify them, and then you can extract the channel you are interested in

**Comment by jice-nospam** [comment]
I'm actually trying to _write_ a single channel file (used as a heightmap). For now I'm adding a second channel filled with zeroes :)

**Comment by johannesvollmer** [comment]
Oh yeah, oopsie. In either case, there is another option: use a custom struct that knows your image data, and make it implement `WritableChannels`. You can pass your struct into a layer, instead of using `SpecificChannels`, like in this example:
https://github.com/johannesvollmer/exrs/blob/49fece069f9bc1e21b252fe0e98311867c38f5d3/examples/5a_write_multiple_layers.rs#L28

**Comment by jice-nospam** [comment]
I tried this way but it seems very complex for what I need, I'll keep my empty channel for now.
Also using globally the exr prelude breaks all the code because you redefine Result. You shouldn't do that!

**Comment by johannesvollmer** [comment]
Yes, if the second channel workaround works for you, it's definitely the nicest code

**Comment by johannesvollmer** [comment]
Would it solve the issue to export a type alias `ExrResult` in the prelude, but keep the original type name `Result` internally?

**Comment by jice-nospam** [comment]
yes I think it would fix the issue

**Comment by jice-nospam** [comment]
oh yes it works ! thanks I didn't know this syntax !

**Comment by johannesvollmer** [comment]
Nice to have it working - but let's keep the issue open, because exrs can be improved nevertheless
